secure/key-manager/src/lib.rs

```diff
@@ -214,14 +214,16 @@ where
     /// increasing.
     fn ensure_timestamp_progress(&mut self) -> Result<(), Error> {
         let current_libra_timestamp = self.libra.libra_timestamp()?;
-        if current_libra_timestamp <= self.last_checked_libra_timestamp {
-            return Err(Error::LivenessError(
-                self.last_checked_libra_timestamp,
-                current_libra_timestamp,
-            ));
+        if let Some(last_checked_libra_timestamp) = self.last_checked_libra_timestamp {
```

Comment:
This feels like we're just avoiding the same race condition as querying too fast and getting the same time. Considering all this does is print out a log... is it worth adding it? I'll leave that to you.

Reply:
Thanks! To address this I've modified the first commit to handle a liveness error in the execution loop (i.e., just log the error rather than killing the process). That way we don't need to worry about this.